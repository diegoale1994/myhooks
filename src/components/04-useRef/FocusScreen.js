import React, { useRef } from 'react'

export const FocusScreen = () => {

    const inputRef = useRef()

    const handleClick = () => {
        inputRef.current.select();
    }

    return (
        <div>
            <h1>Focus Screen</h1>
            <hr />

            <input className="form-control" placeholder="su nombre" ref={inputRef} />
            <br />
            <button className="btn btn-primary" onClick={handleClick}>Focus</button>
        </div>
    )
}
