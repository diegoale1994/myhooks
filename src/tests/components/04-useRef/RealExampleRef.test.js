import { shallow } from "enzyme";
import { RealExampleRef } from "../../../components/04-useRef/RealExampleRef";
import React from 'react';

describe('pruebas en RealExampleRef', () => {
    test('debe hacer match snapshot', () => {
        const wrapper = shallow(<RealExampleRef />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('MultipleCustomHooks').exists()).toBe(false);
    })
    
    test('debe hacer match snapshot', () => {
        const wrapper = shallow(<RealExampleRef />);
        wrapper.find('button').simulate('click');
        expect(wrapper.find('MultipleCustomHooks').exists()).toBe(true);
    })
})
