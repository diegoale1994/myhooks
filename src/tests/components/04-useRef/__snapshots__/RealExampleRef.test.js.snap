// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en RealExampleRef debe hacer match snapshot 1`] = `
<div>
  <h1>
    RealExampleRef
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</div>
`;
